src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexNode.cs

```diff
@@ -69,18 +70,19 @@ private SymbolicRegexNode(SymbolicRegexNodeKind kind, SymbolicRegexNode<TSet>? l
             _set = set;
             _info = info;
             _nullabilityCache = info.StartsWithSomeAnchor && info.CanBeNullable ? new byte[CharKind.ContextLimit] : null;
+#if DEBUG
+            _debugBuilder = builder;
+#endif
         }
 
         /// <summary> Create a new node or retrieve one from the builder _nodeCache</summary>
         private static SymbolicRegexNode<TSet> Create(SymbolicRegexBuilder<TSet> builder, SymbolicRegexNodeKind kind, SymbolicRegexNode<TSet>? left, SymbolicRegexNode<TSet>? right, int lower, int upper, TSet? set, SymbolicRegexInfo info)
         {
-            var key = (kind, left, right, lower, upper, set, info);
+            TSet setOrStartSet = kind == SymbolicRegexNodeKind.Singleton ? set! : ComputeStartSet(builder, kind, left, right);
```

Comment:
Why won't it be null if it's a singleton?

Reply:
I added an assert to check that singleton implies that it's not null. The call sites ensure that, since for singleton the set is what defines what characters the singleton accepts.